src/libbson/src/jsonsl/jsonsl.c

```diff
@@ -457,13 +457,17 @@ jsonsl_feed(jsonsl_t jsn, const jsonsl_char_t *bytes, size_t nbytes)
                 } else if (state->special_flags & JSONSL_SPECIALf_NULL ||
                            state->special_flags & JSONSL_SPECIALf_NAN) {
                    /* previous char was "n", are we parsing null or nan? */
-                   if (CUR_CHAR != 'u') {
+                   bool not_u = CUR_CHAR != 'u';
+                   bool not_a = tolower (CUR_CHAR) != 'a';
```

Comment:
Minor suggestion: rename and redefine `not_u` and `not_a` as `is_u` and `is_a` respectively to avoid possibility of double-negatives in conditions (e.g. `!not_u`).

Reply:
I used the negation here to maintain the intent of the original code, which is "if the char *is not* `c`, unset the bit that corresponds to `c`". Inverting that would be "if the char *is* `c`, do not unset the bit that corresponds to `c`", which also contains a double negative "do not unset". Storing them as variables is just to dedupe the later `&&` check.